pkg/cmd/experimental/syncgroups/grouppruner.go

```diff
@@ -0,0 +1,91 @@
+package syncgroups
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/golang/glog"
+
+	"github.com/openshift/origin/pkg/auth/ldaputil"
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/experimental/syncgroups/interfaces"
+	userapi "github.com/openshift/origin/pkg/user/api"
+)
+
+// GroupPruner runs a prune job on Groups
+type GroupPruner interface {
+	Prune() (errors []error)
+}
+
+// LDAPGroupPruner prunes Groups referencing records on an external LDAP server
+type LDAPGroupPruner struct {
+	// Lists all groups to be synced
+	GroupLister interfaces.LDAPGroupLister
+	// Fetches a group and extracts object metainformation and membership list from a group
+	GroupMemberExtractor interfaces.LDAPMemberExtractor
```

Comment:
Create a new interface who's job it is to detect whether a group exists or not and implement that with the ldap interfaces types